commons/src/main/java/fr/insideapp/sonarqube/apple/commons/antlr/SourceLinesVisitor.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * SonarQube Apple Plugin - Enables analysis of Swift and Objective-C projects into SonarQube.
+ * Copyright © 2022 inside|app (contact@insideapp.fr)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package fr.insideapp.sonarqube.apple.commons.antlr;
+
+
+import fr.insideapp.sonarqube.apple.commons.SourceLine;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SourceLinesVisitor implements ParseTreeItemVisitor {
+
+    private static final Logger LOGGER = Loggers.get(SourceLinesVisitor.class);
+
+    private final int whiteSpaceToken;
+    private final int singleLineCommentToken;
+    private final int multiLineCommentToken;
+    private final int interpolationMultiLineToken;
+    private final int quotedMultiLineTextToken;
+    private final int quotedMultiLineExtendedTextToken;
+
+    public static class Builder
+    {
+        private int whiteSpaceToken = -1;
```

Comment:
Are you sure no token type might be equal to theses values? Wouldn't be safer to make them nullable, to be sure there will never be any collision?

Reply:
We can discuss it, but normally Antlr does not generated token enums with negative integers.
If I switch to Integer / nullable : it will highly increase the complexity of the fillContext method readability (each token has to be tested twice)

Reply:
I agree then. Since the WS is -1, I thought other might also have a negative value. All good then!